api/src/pcapi/routes/backoffice_v3/accounts.py

```diff
@@ -236,6 +249,551 @@ def render_public_account_details(
+            icon="bi-fingerprint",
+            status={
+                "error": _get_status_error(item_status_1517[SubscriptionStep.IDENTITY_CHECK.value]),
+                "success": _get_status_success(item_status_1517[SubscriptionStep.IDENTITY_CHECK.value]),
+                "warning": _get_status_warning(item_status_1517[SubscriptionStep.IDENTITY_CHECK.value]),
+                "active": False,
+                "disabled": False,
+            },
+            id_check_histories=[
+                id_check_history
+                for id_check_history in id_check_histories
+                if id_check_history.type
+                in {
+                    fraud_models.FraudCheckType.UBBLE.value,
+                    fraud_models.FraudCheckType.EDUCONNECT.value,
+                    fraud_models.FraudCheckType.DMS.value,
+                    fraud_models.FraudCheckType.JOUVE.value,
+                }
+            ],
+        ),
+        RegistrationStep(
+            step_id=4,
+            description=SubscriptionStep.HONOR_STATEMENT.value,
+            subscription_item_status=item_status_1517[SubscriptionStep.HONOR_STATEMENT.value],
+            icon="bi-card-checklist",
+            status={
+                "error": _get_status_error(item_status_1517[SubscriptionStep.HONOR_STATEMENT.value]),
+                "success": _get_status_success(item_status_1517[SubscriptionStep.HONOR_STATEMENT.value]),
+                "warning": _get_status_warning(item_status_1517[SubscriptionStep.HONOR_STATEMENT.value]),
+                "active": False,
+                "disabled": False,
+            },
+            id_check_histories=[],
+        ),
+        RegistrationStep(
+            step_id=5,
+            description="Pass 15-17",
+            subscription_item_status=SubscriptionItemStatus.OK.value
+            if user.received_pass_1517
+            else SubscriptionItemStatus.VOID.value,
+            icon="1517",
+            status={
+                "error": False,
+                "success": user.received_pass_1517,
+                "warning": False,
+                "active": False,
+                "disabled": False,
+            },
+            id_check_histories=[],
+        ),
+    ]
+    return steps
+
+
+def _get_tunnel(
+    user: users_models.User, eligibility_history: dict[str, accounts.EligibilitySubscriptionHistoryModel]
+) -> dict:
+    subscription_item_status: dict[str, dict] = {EligibilityType.UNDERAGE.value: {}, EligibilityType.AGE18.value: {}}
+    for key, value in eligibility_history.items():
+        for item in value.subscriptionItems:
+            if key is EligibilityType.UNDERAGE.value:
+                subscription_item_status[EligibilityType.UNDERAGE.value][item.type] = item.status
+            elif key is EligibilityType.AGE18.value:
+                subscription_item_status[EligibilityType.AGE18.value][item.type] = item.status
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** _panik_

Ce bout de code me fait peur (le calcul de `subscription_item_status`)
Certes le code est au moins aussi compliqué que le métier, mais là c'est _trop_ complexe.
Je pense qu'on utilise pas le bon input pour l'output voulu

---

Je pense que ce qu'on cherche est déjà calculé autre part dans le code. Mais je ne suis pas sur de ce qu'on cherche à avoir ici

---

La suite de la fonction en revanche, très clair

Reply:
Pour ceci je ne sais pas trop quoi faire, est-ce qu'on pourrait voir ensemble ?